pkg/sync/syncer.go

```diff
@@ -1974,7 +1974,7 @@ func (s *syncer) processGrantsWithExternalPrincipals(ctx context.Context, princi
 						l.Error("error getting user trait", zap.Any("userPrincipal", userPrincipal))
 						continue
 					}
-					if matchExternalResource.Key == "email" {
+					if strings.EqualFold(matchExternalResource.Key, "email") {
```

Comment:
We're the ones making this key, so I think it's fine to use == here. If it's not "email" (all lower case), something has gone wrong.